### PR TITLE
Sprint #1

### PR DESCRIPTION
**Pull request by jona6689**
Login works for patient users but not for therapists. The login button on the middle of the landing-page is not activated yet - this is an easy fix. Navbar should change according to if a user is logged in. You can login as patient user with: 

Username: hey
Password: hey

Log out should work as well. Give it a go!

The code is a bit messy at the moment and needs refactoring.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+html {
+  overscroll-behavior: none;
+}
+
+br {
+  display: block; /* makes it have a width */
+  content: ""; /* clears default height */
+  margin-top: 5px; /* change this to whatever height you want it */
+}
+
```
Comment (igawaclawska) [inline src/App.css:19]: Please remove the Chat GPT comments

File: src/Components/Login/AdminPage.js [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    fullName: "",
+  });
+  const [professionalUser, setProfessionalUser] = useState({
+    userName: "",
+    password: "",
+    email: "",
+    name: "",
+    numberOfPatients: 0,
+  });
+
```
Comment (igawaclawska) [inline src/Components/Login/AdminPage.js]: Nice use of useState!

File: src/Components/Login/AdminPage.js [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+        }}
+      >
+        <input
+          type="text"
+          name="userName"
+          placeholder="Username"
+          value={patientUser.userName}
+          onChange={(e) => handleChange(e, "patient")}
+          required
+        />
```
Comment (igawaclawska) [inline src/Components/Login/AdminPage.js]: Consider extracting HTML elements such as `<input />,` `<form />`, `<button />` etc., into separate components for better code reusability.

File: src/Components/Navbar/MobileNavigation.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  const location = useLocation(); // Get the current location
+
+  // Function to handle logo click
+  const handleLogoClick = (e) => {
+    if (location.pathname === "/") {
+      e.preventDefault(); // Prevent the default navigation
+      window.location.reload(); // Reload the page
+    }
+  };
```
Comment (igawaclawska) [inline src/Components/Navbar/MobileNavigation.js]: Why do you need to use `window.location.reload()`?

File: src/Components/authService.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+//     PatientUser.set("password", password);
+
+//     await PatientUser.save();
+//     console.log("User registered successfully");
+//     return true;
+//   } catch (error) {
+//     console.error("Error registering user:", error);
+//     return false;
+//   }
+// }
```
Comment (igawaclawska) [inline src/Components/authService.js:17]: Dead code?

File: src/Pages/Chat/Chat.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import "./Chat.css"
+
+function Chat() {
+  return (
+    <p className="woop">
+        YOU HAVE ARRIVED
```
Comment (igawaclawska) [inline src/Pages/Chat/Chat.js:7]: Woop? 😀

File: src/Pages/LandingPage/LandingPageHero.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+              <img src="Images/hands.png" alt="hands" />
+            </div>
+          </div>
+        </div>
+      </section>
+      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1440 120">
+        <path
+          fill="#c0dcf6"
+          d="M0,60C100,90,300,20,480,40C720,60,960,100,1440,40V0H0Z"
+        ></path>
```
Comment (igawaclawska) [inline src/Pages/LandingPage/LandingPageHero.js:43]: You may want to import SVG for better readability

File: src/Components/authService.js [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    const createdUser = await Parse.User.signUp(usernameValue, passwordValue);
+    alert(
+      `Success! User ${createdUser.getUsername()} was successfully created!`
+    );
+    return true;
+  } catch (error) {
+    // signUp can fail if any parameter is blank or failed an uniqueness check on the server
+    alert(`Error! ${error}`);
+    return false;
+  }
```
Comment (igawaclawska) [inline src/Components/authService.js:34]: Nice!

File: src/App.js [hunk 9]
```diff
@@ -1,12 +1,31 @@
...
-    </div>
+    <>
+      <Navbar />
+      <Routes>
+        <Route path="/" element={<LandingPage />} />
+        <Route path="/adminpage" element={<AdminPage />} />
+        <Route path="/loginPatientUser" element={<LoginPatientUser />} />
+        <Route path="/chat" element={<Chat />} />
+      </Routes>
+    </>
```
Comment (igawaclawska) [inline src/App.js]: Good job! 👌

File: src/Components/Navbar/Navbar.css [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+  .nav-whitespace {
+    padding-bottom: 20px;
+  }
+}
+
+.HamburgerMenu {
+  position: absolute;
+  right: 40px;
+  cursor: pointer;
+}
```
Comment (igawaclawska) [inline src/Components/Navbar/Navbar.css:87]: Make sure to use consistent naming convention for your CSS classes and IDs. It's usually recommended to use `kebab-case`.
https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/

File: src/App.js [hunk 11]
```diff
@@ -1,12 +1,31 @@
...
+import AdminPage from "./Components/Login/AdminPage";
+import Parse from "parse";
+import Chat from "./Pages/Chat/Chat";
+
+// Your Parse initialization configuration goes here
+const app_id = process.env.REACT_APP_PARSE_APP_ID; //from back4app --> app settings --> security & keys
+const host_url = process.env.REACT_APP_PARSE_HOST_URL; //form server settings --> host url in the back4app platform
+const javascript_key = process.env.REACT_APP_PARSE_JAVASCRIPT_KEY; //from back4app --> app settings --> security & keys
+Parse.initialize(app_id, javascript_key);
+Parse.serverURL = host_url;
```
Comment (igawaclawska) [inline src/App.js]: Great work on moving the initialization constants into environment variables!

File: src/Components/Login/loginPatientUser.js [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+  const [currentUser, setCurrentUser] = useState(null);
+
+  // Function that will return current user and also update current username
+  const getCurrentUser = async function () {
+    const currentUser = await Parse.User.current();
+    // Update state variable holding current user
+    setCurrentUser(currentUser);
+    return currentUser;
+  };
+
```
Comment (igawaclawska) [inline src/Components/Login/loginPatientUser.js]: 👌

File: src/Components/Navbar/Navbar.js [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+import "./Navbar.css";
+import Parse from "parse";
+
+function Navbar() {
+  const [authenticatedUser, setAuthenticatedUser] = useState(false); // assuming false means not authenticated
+
+  // Check if the user is authenticated
+  useEffect(() => {
+    const currentUserCheck = Parse.User.current();
+
```
Comment (igawaclawska) [inline src/Components/Navbar/Navbar.js:13]: Nice use of useEffect!

**Review by igawaclawska — COMMENTED** [review]
I've added some suggestions for you to consider, but overall nice work!

File: src/Components/Navbar/MobileNavigation.js (hunk 4 above)
Comment (FrederikBjorn) [reply src/Components/Navbar/MobileNavigation.js]: no, it's was for an old idea of navigation, it has now been taken out.

File: src/App.css (hunk 1 above)
Comment (FrederikBjorn) [reply src/App.css:19]: Hi Iga, thanks for all the comments, they are very useful :)

**Pull request merged**